File: kafka-poc/src/main/java/com/study/kafka/model/Message.java
package com.study.kafka.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    private String messageId;
    private String content;
    private MessageFormat format;
    private String topic;
    private long timestamp;
    private Long receivedAt;

    public Message(String content, MessageFormat format) {
        this.content = content;
        this.format = format;
        this.timestamp = Instant.now().toEpochMilli();
    }
}

File: web-client-poc/src/types/message.ts
// Message format enum
export enum MessageFormat {
  JSON = 'JSON',
  TEXT = 'TEXT'
}

// Core Message type
export interface Message {
  messageId: string;
  content: string;
  format: MessageFormat;
  routingKey?: string; // RabbitMQ specific
  topic?: string; // Kafka specific
  timestamp: number; // Epoch milliseconds
  receivedAt?: number; // Epoch milliseconds, set when received
}

// Message send response
export interface MessageSendResponse {
  messageId: string;
  timestamp: number;
  success: boolean;
}

// Bulk message request
export interface BulkMessageRequest {
  count: number;
  messageTemplate: string; // Supports {index} placeholder
  format: MessageFormat;
  delayMs?: number; // Delay between messages
}

// Bulk message response
export interface BulkMessageResponse {
  totalSent: number;
  successCount: number;
  failCount: number;
  durationMs: number;
  throughput: number; // Messages per second
}
